src/components/Game.js

```diff
@@ -0,0 +1,180 @@
+import React, {Component} from 'react'
+import {Alert, AsyncStorage} from 'react-native'
+import styled from 'styled-components/native'
+
+import Board from './Board'
+import Display from './Display'
+
+// The key used to store the high score in AsyncStorage
+const HIGH_SCORE_KEY = 'game:highScore'
+
+// The default time limit defaults to ten ticks
+const timeLimit = 10
+
+// The rows in the game's grid
+const rows = [[0, 1, 2], [3, 4, 5], [6, 7, 8]]
+
+// There are initially nine holes, all which defaults to false
+const initialHoles = [...Array(9)].map(() => false)
+
+const Container = styled.View`
+  flex: 1;
+  justify-content: center;
+  align-items: center;
+  background-color: #000d1a;
+`
+
+const TopBar = styled.View`
+  margin-top: 15px;
+  flex-direction: row;
+  flex: 1.5;
+  align-items: center;
+`
+
+const BottomBar = styled.View`
+  padding-top: 20px;
+  flex: 1;
+  align-items: center;
+`
+
+const StartButton = styled.Button`
+  margin: 20px;
+  padding: 10px;
+`
+
+export default class Game extends Component {
+  state = {
+    highScore: 0,
+    currentScore: 0,
+    timeout: 0,
+    isPlaying: false,
+    holes: initialHoles,
+  }
+
+  async componentDidMount() {
+    try {
+      const highScore = await AsyncStorage.getItem(HIGH_SCORE_KEY)
+
+      if (highScore) {
+        console.log('Retrieved Previous High Score:', highScore)
```

Comment:
This could be `Retrieved current high score`, right?  

Because this state haven't been record the new score.

Reply:
Good point! I'll update the log message.